### PR TITLE
Restructuring CI Workflow

### PR DESCRIPTION
**Pull request by pdet**
This PR has t2o main changes on how we run our CI workflow.

1) **Instead of having one workflow, I broke it into multiple workflows**, the main advantage here is that if a workflow breaks for something unrelated to code changes, we can re-run the broken workflow instead of the whole workflow. 
Since we now have multiple workflows, each of these workflows has a job dependency ( one job that must be executed before running the remaining jobs). In general, I've tried to use the fastest/most reliable job of that workflow, as a sanity check before running the remainder.

2) The Python jobs were frequently running out of time since they were executing many python versions inside the same job, I've added a new matrix parameter on those to alleviate this issue.

I'm happy to make any changes to this suggested workflow structure. 

This is the current suggested structure: (Main job in bold)
1. Main: Basically Main Engine Linux Jobs + Pot-Pourri jobs (e.g., format checker) **(linux-debug)**
2. Linux Release: Jobs that run on the Linux Release Version **(linux-release-64)**
3. NodeJS: NodeJS jobs **(linux-nodejs)**
4. OSX : Main Engine OSX jobs **(xcode-debug)**
5. Python : Python Wheel of all OSs jobs **(linux-python3-9)**
6. R : R jobs **(rstats-linux)**
7. Windows : Main Engine Windows jobs **(win-release-64)**

### CODE REVIEW

**Comment by Mytherin** [comment]
Thanks! Looks great. 

It seems the Python 3.6 builds are failing entirely now, perhaps we should just remove the wheels for that version after all, at least for Linux?

The R Linux package build has been failing for a while now as well, perhaps remove the dependency for now so we can verify that the other R builds work, and we can fix that in a separate PR?

The R failure might be related to an arrow version upgrade as well:

```
2021-11-04T13:32:54.6329997Z ── Error (test_register_arrow.R:256:5): duckdb_register_arrow() performs selection pushdown timestamp type ──
2021-11-04T13:32:54.6331315Z Error in `duckdb_execute(res)`: duckdb_execute_R: Failed to run query
2021-11-04T13:32:54.6333843Z ##[error]Error: Invalid Input Error: arrow_scan: get_next failed(): Invalid: Cannot compare timestamp with timezone to timestamp without timezone, got: timestamp[us] and timestamp[us, tz=UTC]
```

**Comment by pdet** [comment]
Yes, they just released 6.0 in R as well, I'll try to pin it down to 5.0 here and will later make the necessary adjustments to upgrade our arrow support to their new version.

**Comment by pdet** [comment]
@Mytherin It seems to me that one of the reasons that the 3.9 takes too long is because of the arrow tests, maybe we should limit those for one of the machines?

File: .github/workflows/LinuxRelease.yml [hunk 1]
```diff
@@ -0,0 +1,337 @@
+#name: LinuxRelease
+#on: [push, pull_request]
```
Comment (Mytherin) [inline .github/workflows/LinuxRelease.yml:2]: Is this supposed to be all commented out?

**Review by Mytherin — COMMENTED** [review]
Looks great! One more comment, then I think this is ready to merge:

**Comment by pankajp** [comment]
I see Linux Python-3.6 builds got disabled by this PR. Is there some way to make it work or compile on own system for use? Is the failure a fundamental problem or some github actions build image issues (any links to failure logs?). I see mac and windows still have Python 3.6 builds. (I am aware Python 3.6 is getting out of support next month)

**Comment by pdet** [comment]
As you said, Python3.6 support is being discontinued, and our python CI is quite heavy (mostly in Linux because it builds/tests and deploys for like 9 Linux dists).

Maybe we should keep building 3.6 until the next release and then discontinue its support on all platforms. (Any opinions @Mytherin?)

Regarding local builds, I think a
 ``` BUILD_PYTHON=1 make release ```
or a 
``` python tools/pythonpkg/setup.py install ```
should build the py library for you

**Comment by Mytherin** [comment]
`pip install duckdb` should still work for Python 3.6, we have only removed the wheel builds for it for Linux. This is the same as what NumPy and Pandas and such are doing (in fact, the reason we are removing our wheel builds is because they have removed their wheel builds which blows up our CI time). 

> Maybe we should keep building 3.6 until the next release and then discontinue its support on all platforms. (Any opinions @Mytherin?)

Sure, that sounds fine to me.

**Comment by pankajp** [comment]
Thanks @pdet and @Mytherin for the clarification. From what I understand the situation for Linux python 3.6 is not trivial since even if you generate the duckdb wheel, I might have trouble installing the required numpy/pandas builds as they would have to be built from source as well, just like your duckdb CI would have to build them.

**Pull request merged**